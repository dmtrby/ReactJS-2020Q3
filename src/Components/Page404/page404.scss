.error-page {
  background: $black;
  min-height: 100vh;
  display: flex;
  flex-direction: column;

  &__footer {
    margin-top: auto;
  }

  &__header {
    margin-top: 0;
  }

  &__section {
    height: calc(100% - 20rem);
  }

  &__heading {
    text-align: center;
  }

  &__button {
    text-align: center;
  }

  &__404 {
    font-size: 10rem;
    font-weight: bold;
    text-align: center;
    letter-spacing: 2rem;
    color: $primary;
  }
}