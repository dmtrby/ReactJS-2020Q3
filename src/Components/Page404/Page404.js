/* eslint-disable react/forbid-prop-types */
import React from 'react';
import { Link } from 'react-router-dom';

import Logotype from '../Logotype';
import FooterSection from '../FooterSection';
import Heading from '../Base/Heading';
import './page404.scss';

const Page404 = () => (
  <div className="error-page">
    <div className="error-page__header container margin-bottom-2">
      <div className="header__logo-section row start-xs middle-xs padding-top-2">
        <span className="header__logo">
          <Logotype />
        </span>
      </div>
    </div>
    <div className="container error-page__section">
      <div className="flex-column center">
        <div className="col-xs-12 col-lg-10 col-lg-offset-1">
          <Heading classList="error-page__heading" headingLevel={1}>
            Page Not Found
          </Heading>
        </div>
        <div className="error-page__404 col-xs-12 col-lg-10 col-lg-offset-1">404</div>
        <div className="error-page__button col-xs-12 col-lg-10 col-lg-offset-1">
          <Link className="button button--inherit" to="/">Go back home
           
          </Link>
        </div>
      </div>
    </div>
    <div className="error-page__footer">
      <FooterSection />
    </div>
  </div>
);

export default Page404;
