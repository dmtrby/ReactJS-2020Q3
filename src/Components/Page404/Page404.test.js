import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import Page404 from './Page404';

test('404 Page matches the Snapshot', () => {
  const component = renderer.create(
    <Router>
      <Page404 />
    </Router>,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
