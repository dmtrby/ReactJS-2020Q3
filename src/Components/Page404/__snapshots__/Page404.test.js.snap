// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 Page matches the Snapshot 1`] = `
<div
  className="error-page"
>
  <div
    className="error-page__header container margin-bottom-2"
  >
    <div
      className="header__logo-section row start-xs middle-xs padding-top-2"
    >
      <span
        className="header__logo"
      >
        <span
          className="logotype"
        >
          <strong>
            netflix
          </strong>
          roulette
        </span>
      </span>
    </div>
  </div>
  <div
    className="container error-page__section"
  >
    <div
      className="flex-column center"
    >
      <div
        className="col-xs-12 col-lg-10 col-lg-offset-1"
      >
        <h1
          className="error-page__heading"
        >
          Page Not Found
        </h1>
      </div>
      <div
        className="error-page__404 col-xs-12 col-lg-10 col-lg-offset-1"
      >
        404
      </div>
      <div
        className="error-page__button col-xs-12 col-lg-10 col-lg-offset-1"
      >
        <a
          className="button button--inherit"
          href="/"
          onClick={[Function]}
        >
          Go back home
        </a>
      </div>
    </div>
  </div>
  <div
    className="error-page__footer"
  >
    <footer
      className="footer padding-top-1 padding-bottom-1"
    >
      <div
        className="container"
      >
        <div
          className="row center-xs"
        >
          <div
            className="col-xs"
          >
            <div
              className="row center-xs"
            >
              <span
                className="logotype"
              >
                <strong>
                  netflix
                </strong>
                roulette
              </span>
            </div>
          </div>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
